wgsl/index.bs

```diff
@@ -689,84 +682,240 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
 The terms defined in this section express counts of 8-bit bytes.
 
 We will use the following notation:
+* <dfn noexport>AlignOf</dfn>(|T|) is the alignment of host-shareable type |T|.
+* AlignOf(|S|, |M|) is the alignment of member |M| of the host-shareable structure |S|.
+* <dfn noexport>SizeOf</dfn>(|T|) is the size of host-shareable type |T|.
+* SizeOf(|S|, |M|) is the size of member |M| of the host-shareable structure |S|.
+* <dfn noexport>StrideOf</dfn>(|A|) is the element stride of host-shareable array type |A|.
+* <dfn noexport>OffsetOf</dfn>(|S|, |M|) is the offset of member |M| from the start of the host-shareable structure |S|.
+
+
+#### Type Alignment and Size ####  {#type-alignment-and-size}
+
+Every [=host-shareable=] data type has an alignment and size value.
+
+The alignment of a type specifies the byte offsets at which a value of that type
+may be placed relative to the start of the host-shareable structure or array.
+
+The size of a type is the number of contiguous bytes reserved in
+host-shareable memory for the purpose of storing a value of the type.
+The size may include non-addressable padding at the end of the type.
+Consequently, loads and stores of a value might access fewer memory locations
+than value's size.
+
+Both are defined recursively in the following table:
+
+<table class='data'>
+  <caption>
+    Default alignment and strides for host-shareable types<br>
+  </caption>
+  <thead>
+    <tr><th>Host-shareable type |T|
+        <th>[=AlignOf=](|T|)
+        <th>[=SizeOf=](|T|)
+  </thead>
+  <tr><td>[=i32=], [=u32=], or [=f32=]
+      <td>4
+      <td>4
+  <tr><td>`vec2`
+      <td>8
+      <td>8
+  <tr><td>`vec3`
+      <td>16
+      <td>12
+  <tr><td>`vec4`
+      <td>16
+      <td>16
+  <tr><td>mat|X|x|Y| (col-major)
+      <td>[=AlignOf=](vec|Y|)
+      <td>[=SizeOf=](array&lt;vec|Y|, |X|&gt;)
+  <tr><td>`mat2x2`
+      <td>8
+      <td>16
+  <tr><td>`mat3x2`
+      <td>8
+      <td>24
+  <tr><td>`mat4x2`
+      <td>8
+      <td>32
+  <tr><td>`mat2x3`
+      <td>16
+      <td>32
+  <tr><td>`mat3x3`
+      <td>16
+      <td>48
+  <tr><td>`mat4x3`
+      <td>16
+      <td>64
+  <tr><td>`mat2x4`
+      <td>16
+      <td>32
+  <tr><td>`mat3x4`
+      <td>16
+      <td>48
+  <tr><td>`mat4x4`
+      <td>16
+      <td>64
+  <tr><td>struct |S|
+      <td>max([=AlignOf=](S, M<sub>0</sub>), ... , [=AlignOf=](S, M<sub>n</sub>))<br>
+      <td>[=roundUp=]([=AlignOf=](|S|), [=OffsetOf=](|S|, |L|) + [=SizeOf=](|S|, |L|))<br>
+          Where |L| is the last member of the structure
+  <tr><td>[[[=stride=](|S|)]]<br> array<|E|, |N|>
```

Comment:
Maybe change this |S| to a |T| or something to make it less confusing with the |S| used for struct above. I initially thought |S| was struct here.

Reply:
Changed to `Q` as `T` is already defined as a type in the table header. Q doesn't have any pre-associated meanings as far as I'm aware.